etl/src/main/java/org/pih/hivmigration/etl/sql/ProgramMigrator.groovy

```diff
@@ -0,0 +1,220 @@
+
+class ProgramMigrator extends SqlMigrator {
+
+    void migrate() {
+        executeMysql("Create staging table to receive program data", '''
+            create table hivmigration_programs_raw (
+                source_patient_id int PRIMARY KEY,
+                starting_health_center int,
+                health_center int,
+                health_center_transfer_date date,
+                treatment_status varchar(30),
+                treatment_status_date date,
+                art_start_date date,
+                regimen_outcome varchar(20),
+                regimen_outcome_date date,
+                min_visit_date date,
+                max_visit_date date,
+                enrollment_date date,
+                outcome_date date,
+                outcome varchar(255)
+            );
+        ''')
+
+        executeMysql("Create intermediate staging table", '''
+            create table hivmigration_programs (
+                patient_program_id int PRIMARY KEY AUTO_INCREMENT,
+                source_patient_id int,
+                location_id int,
+                enrollment_date date,
+                art_start_date date,
+                outcome_date date,
+                outcome varchar(255)
+            );
+        ''')
+
+        setAutoIncrement("patient_program", "(select max(patient_program_id)+1 from patient_program)")
+        
+        loadFromOracleToMySql('''
+            insert into hivmigration_programs_raw
+                (source_patient_id,
+                starting_health_center,
+                health_center,
+                health_center_transfer_date,
+                treatment_status,
+                treatment_status_date,
+                art_start_date,
+                regimen_outcome,
+                regimen_outcome_date,
+                min_visit_date,
+                max_visit_date)
+            values (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)
+        ''', '''
+            select 	
+                d.patient_id as source_patient_id,
+                d.starting_health_center,
+                d.health_center,
+                d.health_center_transfer_date,
+                d.treatment_status,
+                d.treatment_status_date,
+                r.start_date,
+                o.regimen_outcome,
+                o.regimen_outcome_date,
+                e.min_visit_date,
+                e.max_visit_date
+            from
+                hiv_demographics_real d, hiv_art_regimen_start_date r, hiv_reg_outcome_helper_dates o,
+                (
+                    select 		patient_id, min(encounter_date) as min_visit_date, max(encounter_date) as max_visit_date 
+                    from 		hiv_encounters 
+                    where 		type in ('intake','followup','patient_contact','accompagnateur') 
+                    and			encounter_date is not null
+                    group by 	patient_id
+                ) e
+            where
+                d.TREATMENT_STATUS != 'test' 
```

Comment:
This isn't going to hurt anything, but the hiv_demographics_real view that you are selecting on already does this.